src/coreclr/jit/lower.cpp

```diff
@@ -1825,8 +1837,27 @@ GenTree* Lowering::LowerCallMemmove(GenTreeCall* call)
             BlockRange().Remove(lengthArg);
             BlockRange().Remove(call);
 
+            // Remove all non-user args (e.g. r2r cell)
+            for (CallArg& arg : call->gtArgs.Args())
+            {
+                if (arg.IsArgAddedLate())
+                {
+                    BlockRange().Remove(arg.GetNode());
+                }
+            }
```

Comment:
What if it has side effects?

I think you should just leave this up to DCE, and mark them as unused instead.

Reply:
Addressed